invenio_app_rdm/records_ui/views/decorators.py

```diff
@@ -77,18 +64,48 @@ def view(**kwargs):
     return view
 
 
+def pass_record_or_draft(f):
+    """Decorate to retrieve the record or draft using the record service."""
+    @wraps(f)
+    def view(**kwargs):
+        pid_value = kwargs.get('pid_value')
+        preview = request.args.get('preview')
+        if preview == '1':
```

Comment:
What about another decorator `pass_preview` that sticks `is_preview` into `kwargs` (like `record` is here)? This decorator would make it so that this line becomes:

```suggestion
        preview = kwargs.get('is_preview')
        if is_preview:
```

Then that `preview == '1'` check is centralized which would position us to collapse the conditional urls easily... If that actually works :thinking:

Reply:
I like this idea! Thanks a lot!